rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1044,66 +1084,65 @@ private void processReleaseRequests() {
-                        releaseRequest.getAmount());
+                    "Couldn't build a release BTC tx for <{}, {}>",
+                    releaseRequest.getDestination().toBase58(),
+                    releaseRequest.getAmount());
                 return false;
             }
 
-            // We have a BTC transaction, mark the UTXOs as spent and add the tx
-            // to the release set.
+            BtcTransaction generatedTransaction = result.get().getBtcTx();
+            addPegoutTxToReleaseTransactionSet(generatedTransaction, releaseTransactionSet, releaseRequest);
 
+            // Mark UTXOs as spent
             List<UTXO> selectedUTXOs = result.get().getSelectedUTXOs();
-            BtcTransaction generatedTransaction = result.get().getBtcTx();
-            List<UTXO> availableUTXOs;
-            ReleaseTransactionSet releaseTransactionSet;
+            availableUTXOs.removeAll(selectedUTXOs);
 
-            // Attempt access to storage first
-            // (any of these could fail and would invalidate both
-            // the tx build and utxo selection, so treat as atomic)
-            try {
-                availableUTXOs = getActiveFederationBtcUTXOs();
-                releaseTransactionSet = provider.getReleaseTransactionSet();
-            } catch (IOException exception) {
-                // Unexpected error accessing storage, log and fail
-                logger.error(
-                        String.format(
-                                "Unexpected error accessing storage while attempting to add a release BTC tx for <%s, %s>",
-                                releaseRequest.getDestination().toString(),
-                                releaseRequest.getAmount().toString()
-                        ),
-                        exception
-                );
-                return false;
-            }
+            adjustBalancesIfChangeOutputWasDust(generatedTransaction, releaseRequest.getAmount(), wallet);
 
-            if (activations.isActive(ConsensusRule.RSKIP146)) {
-                Keccak256 rskTxHash = releaseRequest.getRskTxHash();
-                // Add the TX
-                releaseTransactionSet.add(generatedTransaction, rskExecutionBlock.getNumber(), rskTxHash);
-                // For a short time period, there could be items in the release request queue that don't have the rskTxHash
-                // (these are releases created right before the consensus rule activation, that weren't processed before its activation)
-                // We shouldn't generate the event for those releases
-                if (rskTxHash != null) {
-                    // Log the Release request
-                    eventLogger.logReleaseBtcRequested(rskTxHash.getBytes(), generatedTransaction, releaseRequest.getAmount());
-                }
-            } else {
-                releaseTransactionSet.add(generatedTransaction, rskExecutionBlock.getNumber());
+            return true;
```

Comment:
```suggestion
```
Doesn't make much sense this here either

Reply:
The Return Is For The Lamdba in releaseRequestQueue.process() method